tensorrt_llm/commands/serve.py

```diff
@@ -322,29 +343,101 @@ def disaggregated_mpi_worker(config_file: Optional[str], log_level: str):
     )
 
     # Leader ranks will start the trtllm-server using it's own server config
+    # and start a RemoteMPISessionServer to accept MPI tasks
     if is_leader:
+        os.environ[DistaggLauncherEnvs.TLLM_DISTAGG_INSTANCE_IDX] = str(
+            instance_idx)
         server_cfg = disagg_cfg.server_configs[instance_idx]
 
         llm_args, llm_args_extra_dict = get_llm_args(**server_cfg.other_args)
         llm_args = update_llm_args_with_extra_dict(llm_args,
                                                    llm_args_extra_dict)
 
-        mpi_session = MpiCommSession(
-            comm=sub_comm,
-            n_workers=sub_comm.Get_size()) if sub_comm is not None else None
-
-        llm_args["_mpi_session"] = mpi_session
+        _launch_disaggregated_leader(sub_comm, instance_idx, config_file,
+                                     log_level)
 
-        launch_server(host=server_cfg.hostname,
-                      port=server_cfg.port,
-                      llm_args=llm_args)
     else:
+        # Common workers
         with MPICommExecutor(sub_comm) as executor:
             if not is_leader and executor is not None:
                 raise RuntimeError(
                     f"rank{global_mpi_rank()} should not have executor")
 
 
+class DistaggLauncherEnvs(enum.StrEnum):
```

Comment:
Can we use `Disagg` instead of `Distagg` everywhere to be consistent with rest of the code? Thx.

Reply:
Updated, thanks for the reminder.